fix(ci): release-please workflow fails - GitHub Actions cannot create PRs

## Problem

The release-please workflow fails when attempting to create a Release PR after merging commits to `main`.

**Failed run:** https://github.com/bitcraft-apps/klassroom/actions/runs/21080357892/job/60632336593

### Error message

```
##[error]release-please failed: GitHub Actions is not permitted to create or approve pull requests.
- https://docs.github.com/rest/pulls/pulls#create-a-pull-request
```

### What happened

The workflow successfully:
- ✅ Fetched config and manifest
- ✅ Analyzed 68 commits
- ✅ Created branch `release-please--branches--main--components--klassroom`
- ✅ Created commit with version bump (0.2.0 → 0.3.0)
- ❌ Failed to create the Pull Request (permission denied)

## Root Cause

GitHub Actions workflows using `GITHUB_TOKEN` cannot create pull requests by default. This is a repository security setting.

## Solution

### Required fix

Enable the repository setting:

1. Go to **Settings** → **Actions** → **General**
2. Scroll to **Workflow permissions**
3. Check **"Allow GitHub Actions to create and approve pull requests"**
4. Click **Save**

### Optional: Fix tag format warning

The workflow logs show warnings about existing tags:

```
⚠ Found release tag with component '', but not configured in manifest
❯ looking for tagName: klassroom-v0.2.0
```

Existing tags use simple format (`v0.1.0`, `v0.2.0`) but release-please is looking for component-prefixed tags (`klassroom-v0.2.0`).

To ensure consistent tag naming, add `include-component-in-tag` to the config:

```json
{
  "packages": {
    ".": {
      "release-type": "node",
      "bump-minor-pre-major": true,
      "include-component-in-tag": false,
      "extra-files": [...]
    }
  }
}
```

## Reproduction

1. Merge any PR with `feat:` or `fix:` prefix to `main`
2. Observe the release-please workflow fail

## Related

- PR #122 introduced release-please

--- a/release-please-config.json
+++ b/release-please-config.json
@@ -4,6 +4,7 @@
     ".": {
       "release-type": "node",
       "bump-minor-pre-major": true,
+      "include-component-in-tag": false,
       "extra-files": [
         "packages/core/package.json",
         "packages/cli/package.json",

